### DISCUSSION TITLE
Dynamic route doesn't have params

### DISCUSSION THREAD

**Discussion by andriiradkevych** [freshframework/fresh#2217] [General]
**routes/[pageSlug].tsx**

```tsx
import { PageProps } from '$fresh/server.ts';

import PostPage from 'components/Pages/Post/index.tsx';

const Post = async (props: PageProps) => {
  console.log(props.params);

  return <PostPage data={data} />;
};

export default Post;
```

Created dynamic route , and try to get param by url `localhost:3000/something` , but `props.params` return `undefined` , what can be the reason ?

**Comment by marvinhagemeister** [comment]
The `Request` object doesn't have a `.params` property. The [type for async routes](https://fresh.deno.dev/docs/concepts/routes#async-route-components) is:

```tsx
async function MyPage(req: Request, ctx: RouteContext) {}
```

Applying this to your code:

```diff
- import { PageProps } from '$fresh/server.ts';
+ import { RouteContext } from '$fresh/server.ts';

  import PostPage from 'components/Pages/Post/index.tsx';

- const Post = async (props: PageProps) => {
-   console.log(props.params);
+ const Post = async (req: Request, ctx: RouteContext) => {
+   console.log(ctx.params);

  return <PostPage data={data} />;
};

export default Post;
```

It's one of my regrets to match the syntax with middlewares instead of sync components because naturally folks add the `async` keyword in front and expect it to work. It's something I'm hoping to address in a future Fresh release once Preact supports rendering async components. See https://github.com/denoland/fresh/issues/2042
